src/ApiClient.ts

```diff
@@ -58,36 +66,63 @@ export default class ApiClient {
   }
 
   // Raw method for querying the API
-  private _query(req: QueryRequest): ReturnType<typeof MessageApi.Query> {
+  private _query(
+    req: messageApi.QueryRequest
+  ): ReturnType<typeof MessageApi.Query> {
     return retry(
       MessageApi.Query,
-      [req, { pathPrefix: this.pathPrefix }],
+      [req, { pathPrefix: this.pathPrefix, mode: 'cors' }],
       this.maxRetries,
       RETRY_SLEEP_TIME
     )
   }
 
   // Raw method for publishing to the API
-  private _publish(req: PublishRequest): ReturnType<typeof MessageApi.Publish> {
+  private _publish(
+    req: messageApi.PublishRequest
+  ): ReturnType<typeof MessageApi.Publish> {
     return retry(
       MessageApi.Publish,
-      [req, { pathPrefix: this.pathPrefix }],
+      [req, { pathPrefix: this.pathPrefix, mode: 'cors' }],
       this.maxRetries,
       RETRY_SLEEP_TIME
     )
   }
 
   // Raw method for subscribing
   private _subscribe(
-    req: SubscribeRequest,
-    cb: NotifyStreamEntityArrival<Envelope>
-  ): ReturnType<typeof MessageApi.Subscribe> {
-    return retry(
-      MessageApi.Subscribe,
-      [req, cb, { pathPrefix: this.pathPrefix }],
-      this.maxRetries,
-      RETRY_SLEEP_TIME
-    )
+    req: messageApi.SubscribeRequest,
+    cb: NotifyStreamEntityArrival<messageApi.Envelope>
+  ): UnsubscribeFn {
+    let abortController: AbortController
+
+    const doSubscribe = () => {
+      abortController = new AbortController()
+      const startTime = +new Date()
+
+      MessageApi.Subscribe(req, cb, {
+        pathPrefix: this.pathPrefix,
+        signal: abortController.signal,
+        mode: 'cors',
+      }).catch(async (err: any) => {
```

Comment:
This warning seems valid possibly?

Reply:
The reason I never re-throw the error here is because nothing is able to await this promise (since it would block until the stream is done). If I throw, it would lead to an `UnhandledPromiseRejection` error, which is can crash the whole app. So, I just log the error and swallow it.